### PR TITLE
feat: replace hardcoded temp directory paths

### LINKED ISSUE #933: [feature]: replace hardcoding "/tmp/xxx" path
currently we use many `/tmp/xxx` temp directory on db testing, and needs manually delete the dir in the end of test.
It's better to use some crate like [tempfile](https://crates.io/crates/tempfile) to help us.

pros: 
- no need to delete folders manually. programmers may forget to add remove code after testing.
- no need to worry about folder name collision. each test will always use different folder.
- although xline only supports test on linux, it's not bad to add some cross-platform ability.

things to do:
- add crate to `[dev-dependencies]` and replace `/tmp/xxx` path
- for those using `/tmp/xxx` path not in test mod, ex. [crates/xline/src/server/command.rs](https://github.com/xline-kv/Xline/blob/master/crates/xline/src/server/command.rs#L383), use [std::env::temp_dir](https://doc.rust-lang.org/std/env/fn.temp_dir.html) instead. 
    - for this ex. case, we need to define a function to generate snapshot path from uuid. It's not a good idea to use hardcoded string in coding.

### PR DESCRIPTION
**Pull request by cswpy**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
Fixes #933 
* what changes does this pull request make?
This PR replaces the hardcoded tmp path with calls to [tempfile](https://docs.rs/tempfile/latest/tempfile/index.html) crate
* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

**Comment by cswpy** [comment]
I replaced all the hardcoded paths under `/engine`, just wanna make sure this looks right before proceeding to replace the other references.

**Comment by bsbds** [comment]
Hi, please update the PR to resolve CI issues.

File: crates/engine/Cargo.toml [hunk 1]
```diff
@@ -35,3 +35,4 @@ workspace-hack = { version = "0.1", path = "../../workspace-hack" }
 
 [dev-dependencies]
 test-macros = { path = "../test-macros" }
+tempfile = "3"
```
Comment (bsbds) [inline crates/engine/Cargo.toml:38]: Please add a new line at the end.

File: crates/engine/src/rocksdb_engine/mod.rs [hunk 2]
```diff
@@ -756,7 +757,7 @@ mod test {
         };
         let res = engine.apply_snapshot(fake_snapshot, &["not_exist"]).await;
         assert!(res.is_err());
-        fs::remove_dir_all(dir).unwrap();
+        dir.close().unwrap();
     }
 
     #[test]
```
Comment (bsbds) [inline crates/engine/src/rocksdb_engine/mod.rs:763]: The `test_engine_size` test also needs to be updated for consistency.

**Pull request closed without merging**